tests/foreman/api/test_activationkey.py

```diff
@@ -224,37 +177,17 @@ def test_negative_update_1(self, max_content_hosts):
         @Feature: ActivationKey
 
         """
-        try:
-            attrs = entities.ActivationKey().create_json()
-        except HTTPError as err:
-            self.fail(err)
-        activationkey = entities.ActivationKey(id=attrs['id'])
-
-        # Update the activation key with semantically incorrect values.
-        response = client.put(
-            activationkey.path(),
-            {
-                u'unlimited_content_hosts': False,
-                u'max_content_hosts': max_content_hosts
-            },
-            auth=get_server_credentials(),
-            verify=False,
-        )
-        self.assertEqual(
-            response.status_code,
-            httplib.UNPROCESSABLE_ENTITY,
-            status_code_error(
-                activationkey.path(),
-                httplib.UNPROCESSABLE_ENTITY,
-                response
-            ),
-        )
-
-        # Make sure no attributes have changed.
-        new_attrs = activationkey.read_json()
-        for attr in ('unlimited_content_hosts', 'max_content_hosts'):
-            self.assertEqual(attrs[attr], new_attrs[attr])
-        self.assertTrue(new_attrs['unlimited_content_hosts'])
+        act_key = entities.ActivationKey().create()
+        self.assertTrue(act_key.unlimited_content_hosts)
+        with self.assertRaises(HTTPError):
+            entities.ActivationKey(
+                id=act_key.id,
+                max_content_hosts=max_content_hosts,
```

Comment:
Nailgun should pass the data we want. The same way we want to create positive tests we want to create negative ones. This will ensure that the product does not fail when it should not and fail when it should.